CONTRIBUTING.md

```diff
@@ -1,48 +1,219 @@
 
-Please feel free to contribute! While we welcome all contributions to this app, pull-requests that address outstanding Issues *and* have appropriate test coverage for them will be strongly prioritized. In particular, addressing issues that are tagged with the next milestone should be prioritized higher.
+## Pull Requests  
+### Stay scoped
+
+Try to keep your PRs limited to one particular issue, and don't make changes that are out of scope for that issue. If you notice something that needs attention but is out of scope, please [create a new issue](https://github.com/rubyforgood/human-essentials/issues/new).
+
+### In-flight pull requests
+
+If you are so inclined, you can open a draft PR as you continue to work on it. Sometimes we want to get a PR up there and going so that other people can review it or provide feedback, but maybe it's incomplete. This is OK, but if you do it, please tag your PR with `in-progress` label so that we know not to review / merge it.
+
+## Tests 🧪
+### Writing Browser/System/Feature Tests/Specs
+
+Add a test for your change. If you are adding functionality or fixing a  bug, you should add a test!
+
+If you are inexperienced in writing tests or get stuck on one, please reach out for help :). You probably don't need to write new tests when simple re-stylings are done (ie. the page may look slightly different but the Test suite is unaffected by those changes).
+
+If you need to see a browser/system spec run in the browser, you can use the following env variable:
+
+```
+NOT_HEADLESS=true bundle exec rspec
+```
+
+We've added [magic_test](https://github.com/bullet-train-co/magic_test) which makes creating browser specs much easier. It allows you to record actions on the browser running the specs and easily paste them into the spec. You can do this by adding `magic_test` within your system spec:
+```rb
+ it "does some browser stuff" do
+   magic_test
+ end
+```
+and run the spec using this command: `MAGIC_TEST=1 NOT_HEADLESS=true bundle exec rspec <path_to_spec>`
+
+**See videos of it in action [here](https://twitter.com/andrewculver/status/1366062684802846721)**
+
+### Test before submitting pull requests
+- Before submitting a pull request, run all tests and rake tasks with `bundle exec rake` and run lints with `bin/lint`. Fix any broken tests and lints before submitting a pull request.
+- You can run all the tests without rake tasks with `bundle exec rspec`
+- You can run a single test with `bundle exec rspec {path_to_test_name}_spec.rb` or on a specific line by appending `:LineNumber`
+- If you need to skip a failing test, place `pending("Reason you are skipping the test")` into the `it` block rather than skipping with `xit`. This will allow rspec to deliver the error message without causing the test suite to fail.
+
+```ruby
+  it "works!" do
+    pending("Need to implement this")
+    expect(my_code).to be_valid
+  end
+```
+
+## Deployment Process
```

Comment:
This isn't part of the contribution guidelines, no?

Reply:
It's in the README at the bottom in the master repository and issue #3987 said to document the deploy process, so I figured I would put it in CONTRIBUTING. Should I put it back in the README? Or delete it?

Reply:
Let's put it back in the README.

Reply:
resolved by [9cb04c1](https://github.com/rubyforgood/human-essentials/pull/4449/commits/9cb04c13fd8a8a3c52763dc35af527b193b07852)